erc20-wrapper-registry/frontend/pages/index.tsx

```diff
@@ -0,0 +1,233 @@
+      setIsWrongChain(true)
+    }
+  }, [chain])
+
+  return (
+    <main className={styles.main}>
+      <div className={styles.container}>
+        <div className={styles.header}>
+          <h1 className={styles.title}>
+            <span className={styles.gradientText0}>
+              <a href="https://docs.violet.co">VioletID </a>
+            </span>
+            Compliant Token Wrapper Demo
+          </h1>
+
+          <p className={styles.description}>
+            Make your ERC20 token compliant with{' '}
+            <span className={styles.gradientText0}>
+              <a>VioletID</a>
+            </span>
+          </p>
+
+          <div className={styles.connect}>
+            <ConnectWallet
+              dropdownPosition={{
+                side: 'bottom',
+                align: 'center',
+              }}
+            />
+          </div>
+
+          {isWrongChain && status == 'connected' ? (
+            <UnsupportedNetwork />
+          ) : null}
+
+          {isPageLoading ? <LoadingCircles /> : null}
+
+          {!isPageLoading ? (
+            <>
+              {wrapTransaction ? (
+                <SuccessfulWrapTransaction wrapTransaction={wrapTransaction} />
+              ) : (
+                <div>
+                  {status == 'connected' &&
+                  !isWrongChain &&
+                  !isPageLoading &&
+                  !nonCompliantERC20 ? (
+                    <ERC20Input
+                      setTokenInput={setTokenInput}
+                      setNonCompliantERC20={setNonCompliantERC20}
+                      tokenInput={tokenInput}
+                    />
+                  ) : (
+                    <div>
+                      {erc20NotWrapped && readyToWrap && erc20ToBeWrapped ? (
+                        <WrapERC20
+                          erc20ToBeWrapped={erc20ToBeWrapped}
+                          compliantFactoryContract={compliantFactoryContract}
+                          setIsPageLoading={setIsPageLoading}
+                          nonCompliantERC20={nonCompliantERC20}
+                          setWrapTransaction={setWrapTransaction}
+                        />
+                      ) : (
+                        <div>
+                          {erc20InstanceError ? (
+                            <ERC20InstanceError />
+                          ) : (
+                            <div></div>
+                          )}
+                        </div>
+                      )}
+
+                      {compliantErc20 ? (
+                        <DeployedCompliantERC20
+                          foundDeployedWrappedErc20={foundDeployedWrappedErc20}
+                          compliantErc20={compliantErc20}
+                        />
+                      ) : (
+                        <div className={styles.grid}></div>
+                      )}
+                    </div>
+                  )}
+                </div>
+              )}
+            </>
+          ) : null}
```

Comment:
Ternary madness! Hahaha

Suggestion, use intermediary components to create the relevant layers of abstractions:
```
const componentA = (props) => {
  if (!a || !b) return; 
  
  if (c && d)  return <WrapERC20 />
  else return <Whatever /> 
}
```